### ISSUE TITLE
(re-open #384) atlantis_repo_allowlist format for Atlantis and the github_repository_webhooks

### ISSUE THREAD

**Issue by fmudhir** [terraform-aws-modules/terraform-aws-atlantis#394]
## Description

Issue #384 still exists and should not have been closed (the problem cost me many valuable hours). I understand the text by the original poster was hard to decipher but he captured most of whats needed.

- [ x] ✋ I have searched the open/closed issues and my issue is not listed. ( I could not comment/re-open old issue)

## Versions

- Module version [Required]:
v4.2.0
- Terraform version:
v1.5.7
- Provider version(s):
aws 5.0

## Reproduction Code [Required]

NOTE: using `terraform-aws-atlantis` as example repo name

`atlantis_repo_allowlist = ["terraform-aws-atlantis"]`  # from https://github.com/terraform-aws-modules/terraform-aws-atlantis

You only need to start from scratch and populate terraform.tfvars in the github-complete example to reproduce, the documentation is misleading and there's an underlying problem with forwarding atlantis_repo_allowlist to the webhook module.

The env var for the instance needs to be reworked to something like this (sorry I hardcoded the hostname):
```
 # ECS
  atlantis = {
    environment = [
      {
        name  = "ATLANTIS_GH_USER"
        value = var.atlantis_github_user
      },
      {
        name  = "ATLANTIS_REPO_ALLOWLIST"

        # because this module expects a format like:
        #    "github.com/terraform-aws-modules/terraform-aws-atlantis"
        # but the webhook repo module expects only "terraform-aws-atlantis"
        # terraform.tfvars need to have this value: atlantis_repo_allowlist = ["terraform-aws-atlantis"]
        
       # this sort of fixes it, probably need dynamic hostname
       value = join(",", formatlist("github.com/${var.github_owner}/%s", var.atlantis_repo_allowlist)) 
      },
```
+ documentation might needs updating

What makes this problem extra challenging is debugger needs to do terraform destroy to re-apply changes since containers are immutable. So please address this for future developers.

**Comment by bryantbiggs** [comment]
again, what do we need to change here? this looks like users just need to read the documentation for Atlantis, nothing related to this module

**Comment by fmudhir** [comment]
Again this is requesting code change on this repo to the example provided:
terraform-aws-modules/terraform-aws-atlantis/tree/master/examples/github-complete

To change the env var line for allowlist to be as follows.

```
value = join(",", formatlist("github.com/${var.github_owner}/%s", var.atlantis_repo_allowlist))

```
Or (better) changing the module itself to take allowlist in consistent manner to the webhook module. 

I can raise a PR draft if you need more explanation of whats the requested change here.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 4.2.1 :tada:
